//
// Copyright 2018 Asylo authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto2";

// $title: Status
// $overview: Represents an error or OK status.
// $location: https://asylo.dev/docs/reference/proto/util/asylo.status.v1.html
// $front_matter: redirect_from: /docs/reference/proto/util/status.html

package asylo;

// Wire-format representation for a Status object.
message StatusProto {
  // Numeric error code.
  optional int32 code = 1;

  // Detailed error message explaining the error.
  optional string error_message = 2;

  // Name of the error space to which `code` belongs.
  optional string space = 3;

  // The equivalent error code in the Google canonical error space.
  optional int32 canonical_code = 4;

  // Additional payloads in this Status.
  map<string, bytes> payloads = 5;
}
